src/mudlet-lua/lua/geyser/GeyserLabel.lua

```diff
@@ -315,17 +315,15 @@ end
 
 --- Returns the Geyser object associated with the label name
 -- @param label The name of the label to use
-function Geyser.Label:getWindow(label)
-  for i, v in pairs(Geyser.windowList) do
+function Geyser.Label:getWindow(label, cont)
+  cont = cont or Geyser
+  for i, v in pairs(cont.windowList) do
     if v.name == label then
       return v
     end
-
-    -- search down one level to enable nesting in a container
-    for key, val in pairs(v.windowList) do
-      if val.name == label then
-        return val
-      end
+    -- search down one or more levels to enable nesting in a container
+    if self:getWindow(label, v) then 
+      return self:getWindow(label, v) 
```

Comment:
Is it possible to make this infinitely recursive by accident, by putting a child into the grandparent or somesuch?

Reply:
I don't think so. 
It should be exactly the same recursive function as used in show().
The only way (in my opinion) it could be infinitely recursive would be if a container is in it's own container what shouldn't be possible.

Btw I'm thinking about a rewrite of some FlyoutLabel code but I'm not sure I will do that yet.
In my opinion there should be an easier way of doing this.

Reply:
A rewrite sounds good, this code design is a bit messy.

Reply:
Now I got totally rid of that part.
Was simpler then i thought put retesting is needed.